WordPress/Classes/ViewRelated/Post/Revisions/Browser/RevisionDiffsPageManager.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+
+
+protocol RevisionDiffsPageManagerDelegate: class {
+    func currentIndex() -> Int
+    func pageWillScroll(to direction: UIPageViewController.NavigationDirection)
+    func pageDidFinishAnimating(completed: Bool)
+}
+
+
+class RevisionDiffsPageManager: NSObject {
+    var viewControllers: [RevisionDiffViewController] = []
+
+    private unowned var delegate: RevisionDiffsPageManagerDelegate
+
+
+    init(delegate: RevisionDiffsPageManagerDelegate) {
+        self.delegate = delegate
+    }
+
+
+    private func index(of viewController: UIViewController?) -> Int? {
+        return viewControllers.lazy.firstIndex { $0 === viewController }
+    }
+}
+
+
+extension RevisionDiffsPageManager: UIPageViewControllerDataSource {
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
+        return getViewController(at: index(of: viewController),
+                                 direction: .reverse,
+                                 operation: -)
+    }
+
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerAfter viewController: UIViewController) -> UIViewController? {
+        return getViewController(at: index(of: viewController),
+                                 direction: .forward,
+                                 operation: +)
+    }
+
+    private func getViewController(at index: Int?, direction: UIPageViewController.NavigationDirection, operation: (Int, Int) -> Int) -> UIViewController? {
+        guard let index = index else {
```

Comment:
The calling code should deal with the optional index before calling this method.

This is in our swift style guide: [Don't allow optional parameters just to return when they're nil](https://github.com/wordpress-mobile/swift-style-guide#deal-with-optionals-early).